### DISCUSSION TITLE
Gatk Mutect2 input file tumor amd matched normal parameter implementation

### DISCUSSION THREAD

**Discussion by awast** [nextflow-io/nextflow#3745] [Q&A]
Sir,
   I need to run mutect2 on tumor and matched normal 

 Below is my command:

process Mutect2 {
     
   publishDir params.trim, mode : 'copy'

   input:
   tuple val(pair_id), file(tumor_samplefile) , file(tumor_samplefile_bai), file(normal_samplefile), file(normal_samplefile_bai)
   
   
   
   output:
   tuple val(pair_id),file(flr2_file), emit : flr
   tuple val(pair_id),file(pon_file),  emit : pons

   script:
   flr2_file = pair_id + '_f1r2.tar.gz'
  pon_file =  pair_id + '_somatic_variants_PON.vcf' 

  """
  ./gatk --java-options "-Xmx24G"  Mutect2 --native-pair-hmm-threads 20 -R ${params.reference} ${tumor_samplefile} ${normal_samplefile} -pon ${PON_option} -germline-resource ${params.germlineresource} -L ${params.intervalfile} --f1r2-tar-gz $flr2_file -O $pon_file
   
   """
   
}

This is my workflow command

 tumor_bams = tumorsamplenamerun.out.SM_tumorbam 
  tumor_bais = tumorsamplenamerun.out.SM_tumorbam_bai 
  tumor_bam_bai = tumor_bams .join(tumor_bais) | map { tumor_bam, tumor_bai -> [ tumor_bam[1], tumor_bai[2] ] }
  normal_bams = normalsamplenamerun.out.SM_normalbam 
  normal_bais = normalsamplenamerun.out.SM_normal_bai
  normal_bam_bai = normal_bams .join(normal_bais) | map { normal_bam, normal_bai -> [ normal_bam[1], normal_bai[2] ] }
  tn_bambai = tumor_bam_bai .join(normal_bam_bai )| map {tumor_bb, normal_bb -> [ tumor_bb[1], tumor_bb[2], normal_bb[1], normal_bb[2] ] } | Mutect2

}

1. I need to run this for same tumor and matched normal sample i.e. tumor_1_T and normal_1_N but it is taking different samples like
 tumor_2_T and normal_2_N . 
2.also after running this i am getting error like:
Invalid method invocation `call` with arguments: [tumor_1_T, /path/2a/eae4496cf28c566ff8c5e65fd0f392/Tumor_1_T_recal_reads_SM.bam, /media/user/Gallbladder4/trim-recal-dir/2a/eae4496cf28c566ff8c5e65fd0f392/CaGB_1_T_recal_reads_SM.bai] (java.util.ArrayList) on _closure26 type

3. i want to run it in a way that:
/gatk --java-options "-Xmx24G"  Mutect2 --native-pair-hmm-threads 15 -R human_genome -I Tumor1_T_recal_reads_SM.bam  -I Normal1_N_recal_reads_SM.bam -normal NORMAL_1N -pon pon.vcf.gz  -germline-resource somatic-hg38_af-only-gnomad.hg38.vcf.gz -L interval_list --f1r2-tar-gz  f1r2.tar.gz -O tumor_1_somatic_variants_PON.vcf

Please suggest how can i implement

**Comment by complexgenome** [comment]
Hello
Were you able to find solution to it?

**Reply by awast** [reply]
Hi,
yes i have resolved this issue

**Reply by complexgenome** [reply]
Can you please share code/method?
